x-pack/plugins/secops/public/store/local/network/reducer.ts

```diff
@@ -41,6 +57,45 @@ export const initialNetworkState: NetworkState = {
 };
 
 export const networkReducer = reducerWithInitialState(initialNetworkState)
+  .case(updateDnsLimit, (state, { limit, networkType }) => ({
+    ...state,
+    [networkType]: {
+      ...state[networkType],
+      queries: {
+        ...state[networkType].queries,
+        dns: {
+          ...state[networkType].queries!.dns,
+          limit,
+        },
+      },
+    },
+  }))
+  .case(updateDnsSort, (state, { dnsSortField, networkType }) => ({
+    ...state,
+    [networkType]: {
+      ...state[networkType],
+      queries: {
+        ...state[networkType].queries,
+        dns: {
+          ...state[networkType].queries!.dns,
+          dnsSortField,
+        },
+      },
+    },
+  }))
+  .case(updateIsPtrIncluded, (state, { isPtrIncluded, networkType }) => ({
+    ...state,
+    [networkType]: {
+      ...state[networkType],
+      queries: {
+        ...state[networkType].queries,
+        dns: {
+          ...state[networkType].queries!.dns,
```

Comment:
I spent some time studying some of our types such as this one and others like this:

```ts
export interface GenericNetworkModel {
  filterQuery: SerializedFilterQuery | null;
  filterQueryDraft: KueryFilterQuery | null;
  queries: NetworkQueries | null;
}
```
From file: `x-pack/plugins/secops/public/store/local/network/model.ts`

What's the point of having `queries` optionally `null` if we later in the code turn them off?

Example of us turning the check for queries off it looks like?
```ts
export const dnsSelector = () =>
  createSelector(
    selectNetwork,
    network => network.queries!.dns
  );
```

Right now, I am kind of confused about when and why `queries` would be null? It looks like it would never be null and then on the off chance if it ever is `null` it will just blow up?

I kind of stick to my 🔫 guns on if we make a type optional or `null` then that's what it actually can be and there should be `null` guards everywhere around that type.

At the moment our liberal use of `!` is leading to confusion on what is and isn't `null` when and where it is null and why it's not in another function?

I'm finding a lot of correction on pull requests as well as runtime exceptions from confusion around `null` from our excessive use of optional `null` is why I'm bringing it up.

Reply:
This is being addressed in https://github.com/elastic/kibana/pull/33756 -- `queries` within our redux models (starting with this one at least) should no longer be null. Thanks for taking the time to write this up :)